### ISSUE TITLE
Question on Dev Workflow

### ISSUE THREAD

**Issue by curran** [f5/unovis#459]
I notice that the examples under `packages/dev` import things from the Unovis NPM package, for example:

```js
import { VisSingleContainer, VisDonut } from '@unovis/react'
```

My question is, how do Unovis developers test their changes? Is everyone using `npm link` to link the NPM packages to the local source files?

It may be cleaner to leverage NPM workspaces for this, so `npm link` would not be required.

**Comment by rokotyan** [comment]
Can you please describe the problem you're trying to solve? The dev app links the package directly from the source (see the webpack config)

**Comment by curran** [comment]
Yes! Here is a description of the problem, phrased as a user story:

> As a developer new to Unovis, I want to be able to use the VSCode [Go To Definition feature](https://code.visualstudio.com/docs/editor/editingevolved#_go-to-definition), so that I can easily navigate the codebase during development.

Current behavior: this feature doesn't work, because only Webpack knows how to resolve the imports. If symlinks were used, as in NPM workspaces or `npm link`, the Go To Definition feature would work.

## What I Experienced

The problem is a developer experience issue where when you try to follow the imports to their definition in VSCode, like this,

<img width="920" alt="image" src="https://github.com/user-attachments/assets/dbffa089-5006-4e69-8ff5-40453e46dd88">

it resolves to `node_modules`, and then you need to go searching the codebase for the file that matches.

<img width="849" alt="image" src="https://github.com/user-attachments/assets/d8b8631a-af84-4faa-8c25-54ddce2f3224">

Following it to the definition in VSCode lands you in a `.d.ts` file, which is essentially a dead end. You can't easily find the source file where a given export is defined in.

For example, I wanted to inspect the source code that defines `<VisSingleContainer />` so that I could understand what happens to the props, specifically the `height` prop. In order to actually fine the implementation, I need to do a codebase-wide search like this:

<img width="1438" alt="image" src="https://github.com/user-attachments/assets/81e08870-b26d-4f11-9bdf-7e62af75419c">

Then I'm left staring down the results list and clicking through multiple ones to try to figure out if it's the thing that was imported. The only ones that actually match in those search results are these, which are for Vue and Svelte, not React.

<img width="1253" alt="image" src="https://github.com/user-attachments/assets/06b71e3e-d21e-4138-a4a2-194af7d1b9c0">

I spent way too long scrolling up and down that list, only to navigate from the src tree and see if I could find it that way. Eventually I did find it, defined in `packages/react/src/containers/single-container/index.tsx`.

## Suggested Improvement

Let's somehow get the symlinks to work, so that when you follow a definition in VSCode, it takes you straight to the definition, not to a dead-end `.d.ts` file. NPM Workspaces does this by default. So does `npm link`.

I do see this config here in `packages/dev/webpack.config.js`, but that only applies to the build toolchain. It does not create symlinks that VSCode can follow.

I feel like this is worth doing, if only just for my own sanity stepping into this codebase. I will see if `npm link` even works. It could be a rabbit hole...

Thanks for asking for clarification!

**Comment by curran** [comment]
There are some other VSCode features that are also broken, since it can't resolve the imports.

<img width="1053" alt="image" src="https://github.com/user-attachments/assets/99a0945c-227e-4d87-8770-ca75d38f97dc">

**Comment by curran** [comment]
Oh interesting... I wonder if there's some additional VSCode extensions that make it work that I need to install.

I see these issues everywhere in the `dev` package.

**Comment by curran** [comment]
Impacts module resolution errors in VSCode as well.

<img width="909" alt="image" src="https://github.com/user-attachments/assets/e50d59d0-cc69-4e8b-ab27-ec2ed0943de8">

In this case the thing actually is exported and the code that imports it runs fine.

<img width="634" alt="image" src="https://github.com/user-attachments/assets/97a483e3-2ae5-403d-ab55-01c723059a84">

Somehow, the `@unovis/ts` _does_ resolve to the source tree, but it resolve to `lib`, which seems to have generated files.

<img width="963" alt="image" src="https://github.com/user-attachments/assets/08ba9ec2-043e-418c-86c5-8bf4c343e1aa">

Interestingly, the `@unovis/react` import resolves to `node_modules`. I wonder why this is different than `@unovis/ts`...

**Comment by rokotyan** [comment]
@curran Since it points to the built version of `@unovis/ts`, VSCode won't be able to find the export unless you built the library.

We can probably solve that by using TypeScript [Project References](https://www.typescriptlang.org/docs/handbook/project-references.html) or by using [path aliases](https://www.typescriptlang.org/tsconfig/#paths), but it'll require some experimentation.
